server/auth/types/authentication_type.ts

```diff
@@ -159,8 +159,8 @@ export abstract class AuthenticationType implements IAuthenticationType {
       }
 
       // extend session expiration time
-      if (this.config.session.keepalive) {
-        cookie!.expiryTime = Date.now() + this.config.session.ttl;
+      if (this.supportsKeepAlive(request) && this.config.session.keepalive) {
+        cookie!.expiryTime = Math.max(Date.now() + this.config.session.ttl, cookie.expiryTime || 0);
```

Comment:
It may not be strictly necessary to add this line (since only non-JWT implementations will use it), but I think it's fine to make sure keepalive doesn't decrease the existing expiry.

Reply:
Update: changed this to not take the max, since only non-jwt implementations will use it.

Reply:
The addition of `supportsKeepAlive` is great here. This block exists to extend the session for active users which makes sense for basic auth users, but not for OIDC which has its own refresh token workflow or for SAML where the token itself expires.

Adding a note for later, the session extension should also extend to JWT auth but it will need some nuance. A user using JWT auth should be able to extend their session if they are active, but only until the point where the JWT itself expires. We do not want to get to a situation where the session length is greater than the token's expiry date.

Reply:
Resolving this comment due to implementation change. With the new change, JWT sessions will take on the lower of the exp claim and the config's session.ttl, and keep alive is allowed for them. For SAML and OIDC we will be assuming better configurations and those expiry claims will be taken at face value.